### DISCUSSION TITLE
How to disable many engines more efficiently in settings.yml

### DISCUSSION THREAD

**Discussion by nakoo** [searxng/searxng#3696] [Q&A] [answered]
searxng provides many search engines and use them by default. But If I want to disable a lot of engines, I had to manaully declare engine name and add disable option. I feel there is better way to do so, such as disabling everything by default but allow only selected engines. I tried enabled_engines option for this but It just add enabled_engines along with default engines. I'm curious there is better way to hande such cases.

**Comment by return42** [comment, marked as answer]
User can enable/disable all engines in a category.

![image](https://github.com/user-attachments/assets/4de99a0a-a49e-439d-9907-697bd1d2d4b1)

Admins can maintain their own `/etc/searxng/settings.yml` (there is no need to use `use_default_settings: true`)

**Reply by nakoo** [reply]
No, I want to do it in settings.yml

**Reply by return42** [reply]
That's what I suggested ...

> Admins can maintain their own `/etc/searxng/settings.yml` (there is no need to use `use_default_settings: true`) 

and just copy the engines you want from https://github.com/searxng/searxng/blob/98c73010f1f6333d37cd75a75ea924704ba121a2/searx/settings.yml#L322

----

> such as disabling everything by default but allow only selected engines. 

such an option does not exists.

**Comment by Henry-Hiles** [comment]
I'd also love an option like this.

**Comment by mazlo** [comment]
Come across this and saw that meanwhile [there is something](https://docs.searxng.org/admin/settings/settings.html#use-default-settings) like 

```yaml
use_default_settings:
  engines:
    remove:
      ...
```

and 

```yaml
use_default_settings:
  engines:
    keep_only:
      ...
```

**Comment by redge76** [comment]
just copy/paste to chatGTP and ask it to disable everything....
